### PR TITLE
8.0 stock picking reorder lines

### PR DESCRIPTION
**Pull request by damdam-s**
add module 'stock_picking_reorder_lines' migrated to v8 from OCA/sale-workflow

### CODE REVIEW

**Comment by pedrobaeza** [comment]
This already has @gurneyalex :+1: in old PR https://github.com/OCA/sale-workflow/pull/183

**Comment by qtheuret** [comment]
:+1:

File: stock_picking_reorder_lines/stock.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+from openerp import models, api, fields
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+    _order = 'date_expected desc, sequence, id'
+
+    @api.model
+    def _get_sequence(self):
+        last_sequence = 10
```
Comment (pedrobaeza) [inline stock_picking_reorder_lines/stock.py]: This still lacks the query for seeing last assigned sequence.

Comment (qtheuret) [inline stock_picking_reorder_lines/stock.py]: There is no sequence defined for a particular stock picking, so how to compute the next sequence (last sequence +1, +5, +10…) ? If you need to put +1 on the last sequence for each new stock move, the default sequence should be 1, not 10.

Comment (gurneyalex) [inline stock_picking_reorder_lines/stock.py]: @qtheuret it is not needed as we also order by id, so we can work with a single value for default. When reordering with the user interface, all the sequences are recomputed. In the main flow, the sequence is set from the SO.

**Comment by gurneyalex** [comment]
:+1:

File: stock_picking_reorder_lines/stock.py (hunk 1 above)
Comment (pedrobaeza) [inline stock_picking_reorder_lines/stock.py]: That's not true. Let me put you an example to see the problem:
- You add 15 lines. All lines will have sequence = 10.
- You use the widget handle to reorder one of the lines. The lines now have 1, 2, 3... 15 as sequence.
- Add a new line. The new line will appear at the bottom, but the sequence will be 10. You click on Save, and the line will be in the middle of the other lines.

This is a general problem in all the models that use a sequence. A better solution for me would be to put at least a higher number that is difficult to reach (10000 for example), but it's still not perfect.

Comment (gurneyalex) [inline stock_picking_reorder_lines/stock.py]: I would have thought the add at bottom would set the sequence properly. I'm in favor of setting the default to 1000, unless @damdam-s has a better solution.

Comment (damdam-s) [inline stock_picking_reorder_lines/stock.py]: @pedrobaeza , @gurneyalex : solution provided in the last commit

**Comment by pedrobaeza** [comment]
Thanks :+1:

**Comment by qtheuret** [comment]
Thanks :+1:

**Comment by pedrobaeza** [comment]
Travis tests are failing, so I remove my thumb until this get solved.

File: stock_picking_reorder_lines/test/invoice_from_picking.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+-
+  I create a picking with 1 move of 2 products from stock to customer
+-
+  !record {model: stock.picking, id: outgoing_shipment_main_warehouse1}:
+    picking_type_id: stock.picking_type_out
+    invoice_state: '2binvoiced'
+    origin: 'outgoing shipment'
+    partner_id: base.res_partner_6
+    date: !eval "'%s-%s-15' %((datetime.now()-timedelta(days=60)).year, (datetime.now()-timedelta(days=60)).month)"
```
Comment (StefanRijnhart) [inline stock_picking_reorder_lines/test/invoice_from_picking.yml]: This line makes the tests fail:

```
ValueError: time data '2015-6-15' does not match format '%Y-%m-%d %H:%M:%S'
```

**Comment by damdam-s** [comment]
@pedrobaeza update :)

**Comment by pedrobaeza** [comment]
Thanks :+1: 

Merging...

**Comment by damdam-s** [comment]
thanks

**Pull request merged**